src/engraving/dom/engravingitem.cpp

```diff
@@ -1198,6 +1233,114 @@ bool EngravingItem::setProperty(Pid propertyId, const PropertyValue& v)
     return true;
 }
 
+void EngravingItem::manageExclusionFromParts(bool exclude)
+{
+    if (exclude) {
+        std::list<EngravingObject*> links = linkList();
+        for (EngravingObject* linkedObject : links) {
+            if (linkedObject->score() == score()) {
+                continue;
+            }
+            EngravingItem* linkedItem = toEngravingItem(linkedObject);
+            if (linkedItem->selected()) {
+                linkedItem->score()->deselect(linkedItem);
+            }
+            linkedItem->score()->undoRemoveElement(linkedItem, false);
+            linkedItem->undoUnlink();
+        }
+    } else {
+        score()->undoAddElement(this, /*addToLinkedStaves*/ true, /*ctrlModifier*/ false, this);
+    }
+}
+
+void EngravingItem::relinkPropertiesToMaster(PropertyGroup propertyGroup)
+{
+    assert(!score()->isMaster());
+
+    std::list<EngravingObject*> linkedElements = linkListForPropertyPropagation();
+    EngravingObject* masterElement = nullptr;
+    for (EngravingObject* element : linkedElements) {
+        if (element->score()->isMaster()) {
+            masterElement = element;
+            break;
+        }
+    }
+
+    if (!masterElement) {
+        return;
+    }
+
+    const std::set<Pid>& propertiesToRelink = propertyGroup == PropertyGroup::POSITION ? positionProperties()
+                                              : propertyGroup == PropertyGroup::APPEARANCE ? appearanceProperties()
+                                              : textProperties();
+
+    for (Pid propertyId : propertiesToRelink) {
+        PropertyValue masterValue = masterElement->getProperty(propertyId);
+        PropertyFlags masterFlags = masterElement->propertyFlags(propertyId);
+        setProperty(propertyId, masterValue);
+        setPropertyFlags(propertyId, masterFlags);
+    }
+
+    return;
+}
+
+PropertyPropagation EngravingItem::propertyPropagation(EngravingItem* destinationItem, Pid propertyId)
+{
+    assert(destinationItem != this);
+
+    if (propertyLink(propertyId)) {
+        return PropertyPropagation::PROPAGATE; // These properties are always linked, no matter what
+    }
+
+    static std::set<Pid> NEVER_PROPAGATING_PROPERTIES = {
+        Pid::POSITION_LINKED_TO_MASTER,
+        Pid::APPEARANCE_LINKED_TO_MASTER,
+        Pid::TEXT_LINKED_TO_MASTER,
+        Pid::GENERATED,
+        Pid::EXCLUDE_FROM_OTHER_PARTS
+    };
+
+    if (mu::contains(NEVER_PROPAGATING_PROPERTIES, propertyId)) {
+        return PropertyPropagation::NONE;
+    }
+
+    Score* sourceScore = score();
+    Score* destinationScore = destinationItem->score();
+    bool isTextProperty = mu::contains(textProperties(), propertyId);
+
+    if ((isTextProperty && isPropertyLinkedToMaster(propertyId)) || sourceScore == destinationScore) {
```

Comment:
Please check if I added the correct pair of parentheses here (with these added, the meaning stays the same). 

In the lines below, it looks like `sourceScore != destinationScore && ` and  ` || sourceScore == destinationScore` condition a few lines lower are useless (respectively always true and always false, because of the ` || sourceScore == destinationScore` on the line on which I'm commenting right now and the `return` after that. 

(removing such tautological checks is not only good for performance, but also for clarity; whenever I see one of them, I start wondering "is this really what the author meant?", which creates noise.)